boards/components/src/process_console.rs

```diff
@@ -76,12 +77,14 @@ impl<const COMMAND_HISTORY_LEN: usize, A: 'static + Alarm<'static>>
         uart_mux: &'static MuxUart,
         alarm_mux: &'static MuxAlarm<'static, A>,
         process_printer: &'static dyn ProcessPrinter,
+        reboot_function: Option<&'static (dyn Fn() + 'static)>,
```

Comment:
This is fine if we need to get it in in a hurry, but is there a reason not to use generics here to pass the function pointer?

Reply:
I just copied what the capsule driver does, I am happy to improve it with generics.

Reply:
I am not sure how to specify the actual type of the closure in the component.

Reply:
Shouldn't an `Option<fn() -> !>` work? Doesn't even use generics.

Reply:
It works, and we do not actually use a closure.